app/src/main/java/teamagam/ConnectionTest/PingValidator.java

```diff
@@ -0,0 +1,29 @@
+package teamagam.ConnectionTest;
+
+import java.io.IOException;
+
+public class PingValidator implements Validator {
+    private static final String[] URL_PREFIX = {"http://", "https://"};
+
+    @Override
+    public boolean validate(String urlString) {
+        urlString = removePrefix(urlString);
+        try {
```

Comment:
extract the 3 lines of the try to a method:
boolean ping(String url) throws IOException, InterruptedException {
  ...
}